pvlib/snowcoverage.py

```diff
@@ -0,0 +1,229 @@
+    implemented in SAM [2].
+
+    Parameters
+    ----------
+    snow_data : numeric
+        Time series data on either snowfall or ground snow depth. The type of
+        data should be specified in snow_data_type. The original model was
+        designed for ground snowdepth only. (cm/hr or cm)
+
+    time_step_hours: float
+        Period of the data in hours. (hours between data points)
+
+    snow_data_type : string
+        Defines what type of data is being passed as snow_data. Acceptable
+        values are "snowfall" and "snow_depth". "snowfall" will be in units of
+        cm/hr. "snow_depth" is in units of cm.
+
+    Returns
+    ----------
+    fully_covered_mask : boolean array
+        True where the snowfall exceeds the defined threshold to fully cover
+        the panel. False elsewhere.
+
+    References
+    ----------
+    [1] Marion, B.; Schaefer, R.; Caine, H.; Sanchez, G. (2013).
+    “Measured and modeled photovoltaic system energy losses from snow for
+    Colorado and Wisconsin locations.” Solar Energy 97; pp.112-121.
+    '''
+    if snow_data_type == "snow_depth":
+        prev_data = snow_data.shift(1)
+        prev_data[0] = 0
+        snowfall = snow_data - prev_data
+    elif snow_data_type == "snowfall":
+        snowfall = snow_data
+    else:
+        raise ValueError('snow_data_type was not specified or was not set to a'
+                         'valid option (snowfall, snow_depth).')
+
+    time_adjusted = snowfall / time_step_hours
+    fully_covered_mask = time_adjusted >= 1
+    return fully_covered_mask
+
+
+def snow_coverage_model(snow_data, snow_data_type,
+                        poa_irradiance, temperature, surface_tilt,
+                        time_step_hours=1, m=-80, sliding_coefficient=1.97):
+    '''
+    Calculates the fraction of a module covered by snow at every time step
+    following the same model first described in [1] and later
+    implemented in SAM [2]. Currently only implemented for fixed tilt systems
+    but could be generalized for tracking systems.
+
+    Parameters
+    ----------
+    snow_data : numeric
+        Time series data on either snowfall or ground snow depth. The type of
+        data should be specified in snow_data_type. The original model was
+        designed for ground snowdepth only. (cm/hr or cm)
+
+    snow_data_type : string
+        Defines what type of data is being passed as snow_data. Acceptable
+        values are "snowfall" and "snow_depth". "snowfall" will be in units of
+        cm/hr. "snow_depth" is in units of cm.
+
+    poa_irradiance : numeric
+        Total in-plane irradiance (W/m^2)
+
+    temperature : numeric
+        Ambient air temperature at the surface (C)
+
+    surface_tilt : numeric
+        Surface tilt angles in decimal degrees. Tilt must be >=0 and
+        <=180. The tilt angle is defined as degrees from horizontal
+        (e.g. surface facing up = 0, surface facing horizon = 90).
+
+    time_step_hours: float
```

Comment:
Could the time_step be obtained from the index of snow_data?

Reply:
Could be but maybe not always? I thought it might be better to allow it as an input.

Reply:
but by having a `time_step` input, equal time steps are assumed. If `snow_data` is a Series, then possibly unequal time steps could be accommodated.